scripts/s3manage.py

```diff
@@ -0,0 +1,522 @@
+                    ACL='public-read')
+            except botocore.exceptions.ClientError as e:
+                print('Cannot copy {}: {}'.format(j, e))
+        # TODO Adapt get_url to final URLs
+        print('\nPlease check it on {}'.format(get_url()))
+        print('And {}'.format(get_url(key_name='src/index.html')))
+    else:
+        print('Aborting activation of version {}'.format(s3_path))
+        sys.exit(1)
+
+
+def init_connection(bucket_name, profile_name):
+    try:
+        session = boto3.session.Session(profile_name=profile_name)
+    except botocore.exceptions.ProfileNotFound as e:
+        print('You need to set PROFILE_NAME to a valid profile name in $HOME/.aws/credentials')
+        print(e)
+        sys.exit(1)
+    except botocore.exceptions.BotoCoreError as e:
+        print('Cannot establish connection. Check you credentials %s.' % profile_name)
+        print(e)
+        sys.exit(1)
+
+    s3client = session.client('s3', config=boto3.session.Config(signature_version='s3v4'))
+    s3 = session.resource('s3', config=boto3.session.Config(signature_version='s3v4'))
+
+    bucket = s3.Bucket(bucket_name)
+    return (s3, s3client, bucket)
+
+
+def exit_usage(cmd_type):
+    print('Missing one arg for %s command' % cmd_type)
+    usage()
+    sys.exit(1)
+
+
+def parse_arguments(argv):
+    if len(argv) < 2:
+        exit_usage('UNKNOWN')
+
+    cmd_type = str(argv[1])
+    deploy_target = 'int'
+
+    supported_cmds = ('upload', 'list', 'info', 'activate', 'delete')
+    if cmd_type not in supported_cmds:
+        print('Command %s not supported' % cmd_type)
+        usage()
+        sys.exit(1)
+
+    if cmd_type == 'upload' and len(argv) != 4:
+        exit_usage(cmd_type)
+    elif cmd_type == 'list' and len(argv) != 2:
+        exit_usage(cmd_type)
+    elif cmd_type == 'info' and len(argv) != 3:
+        exit_usage(cmd_type)
+    elif cmd_type == 'activate' and len(argv) != 4:
+        exit_usage(cmd_type)
+    elif cmd_type == 'delete' and len(argv) != 3:
+        exit_usage(cmd_type)
+
+    base_dir = os.getcwd()
+    if cmd_type == 'upload':
+        base_dir = os.path.abspath(argv[2])
+        if not os.path.isdir(base_dir):
+            print('No code found in directory %s' % base_dir)
+            sys.exit(1)
+
+    if cmd_type in ('activate', 'upload'):
+        deploy_target = argv[3]
+        if deploy_target not in ('int', 'prod'):
+            print('%s is not a valid deploy target' % deploy_target)
+            sys.exit(1)
+
+    s3_path = None
+    if cmd_type in ('info', 'activate', 'delete'):
+        s3_path = argv[2]
+        if s3_path.endswith('/'):
+            s3_path = s3_path[:len(s3_path) - 1]
+        if s3_path.count('/') != 2:
+            print('Bad version definition')
+            usage()
+            sys.exit(1)
+
+    bucket_name_env = 'S3_MF_GEOADMIN3_%s' % deploy_target.upper()
+    # TODO Remove me when switch has been performed
+    bucket_name_env = 'S3_MF_GEOADMIN3_INFRA'
```

Comment:
Sure that's the idea. We can't test anything as s3 native addresses are blocked at the moment and I need Andrea to make the necessary Varnish changes.